### PR TITLE
added log statements and updated .gitignore

### PR DESCRIPTION
**Pull request by n-aggarwal**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+debug
```
Comment (friendlymatthew) [inline .gitignore:2]: why do we need the debug in the gitignore? @n-aggarwal

Comment (n-aggarwal) [reply .gitignore:2]: I had over 4000 uncommitted files before I added debug in the gitignore. I think this folder was created during cargo build. That's why I have that here

Comment (TransientError) [reply .gitignore:2]: Cargo creates a `target` directory, which will often contain like `debug`, `release` and any other compilation modes you add to your build. Would recommend adding `target` instead.

File: julia/src/api/bond_routes.rs [hunk 2]
```diff
@@ -15,13 +16,15 @@ pub enum ApiError {
 impl Reject for ApiError {}
 
 pub fn routes(session: Arc<Session>) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    info!("Function \"routes\" called.");
```
Comment (TransientError) [inline julia/src/api/bond_routes.rs:19]: While it's reasonable to use `info` here, I'd suggest `trace`. I believe the purpose of `trace` is to trace a request through code execution. Like a generic example might be figuring out which if/else branch did the code execute. 

That being said all sorts of people/teams have their own ideas of how to use each logging level etc, so it's not like info is wrong per se. I just wanted to share my opinion.

File: julia/src/api/bond_routes.rs [hunk 3]
```diff
@@ -30,10 +33,15 @@ pub fn create_bond_route(session: Arc<Session>) -> impl Filter<Extract = impl wa
...
+    info!("Function \"handle_create_bond\" called.");
     match form_bond(session.clone(), bond).await {
-        Ok(_) => Ok(warp::reply::with_status("Bond created successfully!", StatusCode::CREATED)),
+        Ok(_) => {
+            info!("Bond created successfully!");
+            Ok(warp::reply::with_status("Bond created successfully!", StatusCode::CREATED))
+        }
         Err(e) => {
             eprintln!("Error occurred while creating bond: {:?}", e);
+            info!("Error occurred while creating bond: {:?}", e);
```
Comment (TransientError) [inline julia/src/api/bond_routes.rs:44]: There is not really a point to `eprintln` and then also log a statement. I'd suggest you do one or the other.

**Comment by TransientError** [comment]
Just a general tip but if you're interested in learning more about tracing, this could be an interesting place to apply a [`Span`](https://docs.rs/tracing/latest/tracing/span/index.html) instead of of just plain events. E.g. like you could have a "create bond" span and similar for the other apis.

**Pull request closed without merging**